### DISCUSSION TITLE
How to add tracks with GFF file format?

### DISCUSSION THREAD

**Discussion by GLB-KD** [GMOD/jbrowse-components#1404] [General] [answered]
I am following the steps as mentioned in the following link: [https://jbrowse.org/jb2/docs/quickstart_web/](url) to add assembly and tracks. I have uploaded the .gff file to the local instance. How can I add a track for .gff format? 

For assembly, we need to create a fasta-index (fai) file. Do we need to follow any such steps? It shows that the track type is not supported.

**Comment by garrettjstevens** [comment, marked as answer]
JBrowse 2 currently supports indexed GFF3. If your file is a valid and sorted [gff3](https://m.ensembl.org/info/website/upload/gff3.html), you can index it by running:

```
bgzip my.gff3
tabix my.gff3.gz
```

This will create the files `my.gff3.gz` and `my.gff3.gz.tbi`, which can be opened in JBrowse 2.

**Reply by GLB-KD** [reply]
Thanks. I did the same thing and found that there was a sorting issue with the GFF file.  
To sort the files, I used the following command:
`sort -k1,1 -k4,4n myfile.gff > myfile.sorted.gff`
`(OR)`
`gt gff3 -sortlines -tidy -retainids myfile.gff > myfile.sorted.gff`

**Comment by cmdcolin** [comment]
technically this would be a good thing to add to our quickstart. A lot of GFF need to be sorted also before being properly gzipped...maybe discuss in meeting

**Reply by GLB-KD** [reply]
Sure. Please add the steps to the quickstart. Also, as you mentioned GFF needs to be sorted. If you can add the steps to sort the GFF file, it would be great.
